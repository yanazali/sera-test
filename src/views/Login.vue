<template>
  <div>
    <AuthForm
      @btnSubmit="handleBtnSubmit"
      title="Login"
      labelBtn="Login"
      linkTo="register"
      labelLinkTo="Register"
    />
  </div>
</template>

<script>
import { mapActions } from "vuex";
import AuthForm from "@/components/AuthForm.vue";

export default {
  metaInfo: {
    titleTemplate: "Login",
  },
  components: {
    AuthForm,
  },
  methods: {
    ...mapActions("auth", ["login"]),
    handleBtnSubmit: function (form) {
      this.login({
        email: form.email,
        password: form.password,
      })
        .then(() => {
          this.$swal({
            text: "Login Berhasil",
            icon: "success",
            confirmButtonText: "Ok",
          });
          this.$router.push("/");
        })
        .catch((err) => {
          this.$swal({
            text: err.response.data.error,
            icon: "error",
            confirmButtonText: "Ok",
          });
          console.log(err.response.data);
        });
    },
  },
  created() {
    let token = sessionStorage.getItem("token");
    if (token) {
      this.$router.push("/");
    }
  },
};
</script>
