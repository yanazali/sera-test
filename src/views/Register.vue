<template>
  <div>
    <AuthForm
      @btnSubmit="handleBtnSubmit"
      title="Register"
      labelBtn="Register"
      linkTo="login"
      labelLinkTo="Back to Login"
    />
  </div>
</template>

<script>
import { mapActions } from "vuex";
import AuthForm from "@/components/AuthForm.vue";

export default {
  metaInfo: {
    titleTemplate: "Register",
  },
  components: {
    AuthForm,
  },
  methods: {
    ...mapActions("auth", ["register"]),
    handleBtnSubmit: function (form) {
      this.register({
        email: form.email,
        password: form.password,
      })
        .then(() => {
          this.$swal({
            text: "Register Berhasil",
            icon: "success",
            confirmButtonText: "Ok",
          });
          this.$router.push("/login");
        })
        .catch((err) => {
          this.$swal({
            text: err.response.data.error,
            icon: "error",
            confirmButtonText: "Ok",
          });
          console.log(err.response.data);
        });
    },
  },
  created() {
    let token = sessionStorage.getItem("token");
    if (token) {
      this.$router.push("/");
    }
  },
};
</script>

