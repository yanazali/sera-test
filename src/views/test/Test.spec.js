import { shallowMount } from '@vue/test-utils';
import Login from "@/views/Login.vue";

describe('Login Page', () => {
  it('should render correctly', () => {
    const wrapper = shallowMount(Login);
    expect(wrapper.element).toMatchSnapshot();
  });
});

import Register from "@/views/Register.vue";

describe('Register Page', () => {
  it('should render correctly', () => {
    const wrapper = shallowMount(Register);
    expect(wrapper.element).toMatchSnapshot();
  });
});

// import BlogPostsList from "@/views/Crud/BlogPostsList.vue";

// describe('Blog Posts List Page', () => {
//   it('should render correctly', () => {
//     const wrapper = shallowMount(BlogPostsList);
//     expect(wrapper.element).toMatchSnapshot();
//   });
// });

