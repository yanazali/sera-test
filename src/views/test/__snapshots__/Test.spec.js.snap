// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should render correctly 1`] = `
<div>
  <authform-stub
    labelbtn="Login"
    labellinkto="Register"
    linkto="register"
    title="Login"
  />
</div>
`;

exports[`Register Page should render correctly 1`] = `
<div>
  <authform-stub
    labelbtn="Register"
    labellinkto="Back to Login"
    linkto="login"
    title="Register"
  />
</div>
`;
